tests/lib/prepare.sh

```diff
@@ -775,6 +775,39 @@ uc20_build_corrupt_kernel_snap() {
     rm -rf "$REPACKED_DIR"
 }
 
+uc_write_bootstrap_wrapper() {
+    local SKELETON_PATH="$1"
+    local INJECT_ERR="${2:-false}"
+
+    cp -a /usr/lib/snapd/snap-bootstrap "$SKELETON_PATH"/usr/lib/snapd/snap-bootstrap.real
+    cat <<'EOF' >"$SKELETON_PATH"/usr/lib/snapd/snap-bootstrap
+#!/bin/sh
+set -eux
+if [ "$1" != initramfs-mounts ]; then
+    exec /usr/lib/snapd/snap-bootstrap.real "$@"
+fi
+beforeDate="$(date --utc '+%s')"
+/usr/lib/snapd/snap-bootstrap.real "$@"
+if [ -d /run/mnt/data/system-data ]; then
+    touch /run/mnt/data/system-data/the-tool-ran
```

Comment:
I think we should clean that up. On 22+, it is not called `the-tool` anymore. And since we have systemd re-execution in 22+ we can actually ask systemd directly if and when it was run.

Reply:
And of course, in a follow up PR